FlowCal/plot.py

```diff
@@ -1061,6 +1302,1271 @@ def hist1d(data_list,
+    num_y_bins : int, optional
+        Number of bins to bin population members into along the y-axis.
+        Ignored if `y_bin_edges` is specified.
+    y_bin_edges : sequence of scalars, sequence of sequence of scalars, or
+                  mapping to sequence of scalars, optional
+        Bin edges used to bin population members along the y-axis. If a
+        mapping is specified, each violin's `y_bin_edges` are searched for in
+        the following key order: the violin's position, 'data'. Min, max, and
+        zero `y_bin_edges` can be specified via the 'min', 'max', and
+        'logx_zero' keys, respectively. If `y_bin_edges` is a sequence of
+        sequence of scalars, min, max, and zero violins cannot be used. If
+        not specified, `y_bin_edges` is calculated linearly (`yscale` ==
+        'linear') or logarithmically (`yscale` == 'log') to span `ylim` using
+        `num_y_bins`.
+    upper_trim_fraction : float, sequence of floats, or mapping to float,
+                          optional
+        Fraction of members to trim (discard) from the top of the violin (for
+        aesthetic purposes). If a mapping is specified, each violin's
+        `upper_trim_fraction` is searched for in the following key order: the
+        violin's position, 'data'. Min, max, and zero `upper_trim_fraction`
+        can be specified via the 'min', 'max', and 'logx_zero' keys,
+        respectively. If `upper_trim_fraction` is a sequence of sequence of
+        floats, min, max, and zero violins cannot be used.
+    lower_trim_fraction : float, sequence of floats, or mapping to float,
+                          optional
+        Fraction of members to trim (discard) from the bottom of the violin
+        (for aesthetic purposes). If a mapping is specified, each violin's
+        `lower_trim_fraction` is searched for in the following key order: the
+        violin's position, 'data'. Min, max, and zero `lower_trim_fraction`
+        can be specified via the 'min', 'max', and 'logx_zero' keys,
+        respectively. If `lower_trim_fraction` is a sequence of sequence of
+        floats, min, max, and zero violins cannot be used.
+    violin_width_to_span_fraction : float, optional
+        Fraction of the x-axis span that a violin should span. Ignored if
+        `violin_width` is specified.
+    violin_kwargs : mapping, sequence of mappings, mapping to mappings,
+                    optional
+        Keyword arguments passed to the plt.fill_betweenx() command that
+        illustrates each violin. If a mapping to mappings is specified, each
+        violin's kwargs are searched for in the following key order: the
+        violin's position, 'data'. Min, max, and zero kwargs can be specified
+        via the 'min', 'max', and 'logx_zero' keys, respectively. If
+        `violin_kwargs` is a sequence of mappings, min, max, and zero violins
+        cannot be used. Default = {'facecolor':'gray', 'edgecolor':'black'}.
```

Comment:
This needs examples. It's a little hard to understand.

Reply:
Agreed.

Several parameters behave similarly in this regard. I'm not sure if it's better to make separate examples for all of them or to reorganize the docstring and consolidate like parameters. E.g. adding a "Per-Violin Parameters" section and explaining the different ways they can be specified.

Reply:
The problem with the  "Per-Violin Parameters" idea is that I don't know how `sphinx` would parse it when generating the documentation. I'm OK with having examples for all of them, even if that means that there is some repetition.

Reply:
`sphinx` parses "Other Parameters" sections for many functions in [the `plot` module](https://flowcal.readthedocs.io/en/latest/reference/FlowCal.plot.html). Is there any reason to believe a "Per-Violin Parameters" section (or an "Advanced Parameters" section for `min_data`, `max_data`, and `log_zero_data`, as discussed previously) would be treated any differently?

Reply:
There is a documented "Other Parameters" section in the [`numpydoc` specification](https://numpydoc.readthedocs.io/en/latest/format.html#sections).